script.js

```diff
@@ -49,37 +86,61 @@ function validateForm() {
         return;
     }
 
+    // Create user object
+    const newUser = { username, fullName, age, email, password };
 
-    //localStorage.setItem("username", username);
-    //localStorage.setItem("fullName", fullName);
-    //localStorage.setItem("age", age);
-    //localStorage.setItem("email", email);
-    //localStorage.setItem("password", password);
-    //localStorage.setItem("repeatPassword", repeatPassword);
-
-    let userRecords = new Array();
-    userRecords = JSON.parse(localStorage.getItem("users")) ? JSON.parse(localStorage.getItem("users")) : []
-    if (userRecords.some((v) => {
-        return v.email == email
-
-    })) {
-        alert("This email already exists!");
-
-    } else {
-        userRecords.push({
-            "username": username,
-            "fullName": fullName,
-            "age": age,
-            "email": email,
-            "password": password,
-            "repeatPassword": repeatPassword
-        })
-        localStorage.setItem("users", JSON.stringify(userRecords));
-        // Redirect to the success page
-        window.location.href = 'success.html';
-    }
+    // Save user data to local storage
+    const users = JSON.parse(localStorage.getItem('users')) || [];
+    users.push(newUser);
+    localStorage.setItem('users', JSON.stringify(users));
 
+    // Clear the form
+    document.getElementById('registrationForm').reset();
+
+    // Update user listbox
+    updateListBox();
+
+    alert('User registered successfully!');
+}
 
+// Function to update the user selection listbox
+function updateListBox() {
+    const userList = document.getElementById('userList');
+    userList.innerHTML = '';
+
+    const users = JSON.parse(localStorage.getItem('users')) || [];
+    users.forEach(user => {
+        const option = document.createElement('option');
+        option.value = user.username;
+        option.text = user.username;
+        userList.add(option);
+    });
+}
```

Comment:
modify this code so user can be selected when there is only one user in the list